jupyter_server/extension/application.py

```diff
@@ -108,6 +108,10 @@ def _default_extension_name(self):
 
     INVALID_EXTENSION_NAME_CHARS = [' ', '.', '+', '/']
 
+    def _config_file_name_default(self):
```

Comment:
There is some logic in `jupyter_core.JupyterApp` for [replacing hyphens with underscores](https://github.com/jupyter/jupyter_core/blob/b129001bb88abf558df5dfab791a5aeeba79e25c/jupyter_core/application.py#L113-L116). We should probably maintain that logic here in case extension names have hyphens.